src/_pytest/logging.py

```diff
@@ -72,24 +72,80 @@ class PercentStyleMultiline(logging.PercentStyle):
     formats the message as if each line were logged separately.
     """
 
+    def __init__(self, fmt, auto_indent):
+        super().__init__(fmt)
+        self._auto_indent = auto_indent
+
     @staticmethod
     def _update_message(record_dict, message):
         tmp = record_dict.copy()
         tmp["message"] = message
         return tmp
 
+    @staticmethod
+    def _get_auto_indent(record, default_auto_indent) -> int:
+        """Determines the current auto indentation setting
+
+        Specify auto indent behavior (on/off/fixed) by passing in
+        extra={"auto_indent": [value]} to the call to log.log() or
+        using a --log-auto-indent [value] command line or config option.
+        Default behavior is auto-indent on.
+
+        Using the string "True" or "on" or the boolean True as the value
+        turns auto indent on, using the string "False" or "off" or the
+        boolean False or the int 0 turns it off, and specifying a
+        positive integer fixes the indentation position to the value
+        specified.
+
+        Any other values for the option are invalid, and will silently be
+        converted to "on".
+        """
+        out_auto_indent = -1
+        if hasattr(record, "auto_indent"):
+            # passed in from the "extra={}" kwarg on the call to log.log()
+            in_auto_indent = record.auto_indent
+        elif default_auto_indent is not None:
+            in_auto_indent = default_auto_indent
+        else:
+            in_auto_indent = -1
+
+        if type(in_auto_indent) is int:
+            out_auto_indent = in_auto_indent
+        if type(in_auto_indent) is str:
+            if in_auto_indent.lower() == "true" or in_auto_indent.lower() == "on":
+                out_auto_indent = -1
+            elif in_auto_indent.lower() == "false" or in_auto_indent.lower() == "off":
+                out_auto_indent = 0
+            else:
+                try:
+                    out_auto_indent = int(in_auto_indent)
+                except ValueError:
+                    out_auto_indent = -1
+        elif type(in_auto_indent) is bool:
+            if in_auto_indent:
+                out_auto_indent = -1
+            else:
+                out_auto_indent = 0
```

Comment:
Please move this code to a new method and call it in `__init__`, where you assign the return value to self._auto_indent and in `_get_auto_indent`. You can then change the implemenation of _get_auto_indent to 

```
def _get_auto_indent(self, .....):
        if hasattr(record, "auto_indent"):
            # passed in from the "extra={}" kwarg on the call to log.log()
            out_auto_indent = self.TODO_NEW_METHOD(record.auto_indent)
        else:
            out_auto_indent = self._auto_indent
        return out_auto_indent
```
Since the above method is very simple you can also move it directly to `def format`.

Reply:
I followed this advice in the latest iteration, including calling this code for the config option in the `__init__`, but I did not need to create a new method. Having moved the code to `def format` as you suggested, this left all the code that would have been moved to a new method already isolated in `def _get_auto_indent`, so this accomplished what I believe you had intended.